### ISSUE TITLE
Assembly.Load executed in collectible context loads into default one

### ISSUE THREAD

**Issue by janvorli** [dotnet/coreclr#22270]
When Assembly.Load is executed from a code running in a collectible AssemblyLoadContext, it loads the assembly into the default context instead of the current context. However, when the Assembly.Load is executed from a code running in a **non**-collectible AssemblyLoadContext, it loads the assembly correctly into the current context.

@vitek-karas and I were looking at the code of `AssemblyNative::Load` and we've immediately spotted the cause. There is a condition like this:
```C++
        // Shared or collectible assemblies should not be used for the parent in the
        // late-bound case.
        if (pRefAssembly && (!pRefAssembly->IsCollectible()))
        {
            pParentAssembly= pRefAssembly->GetDomainAssembly();
        }
```

It is possible that the `!pRefAssembly->IsCollectible()` is obsolete or that it was supposed to cover reflection emitted collectible assemblies.

**Comment by janvorli** [comment]
I've verified that removing the IsCollectible check in the code shown above fixes the issue.
However, we need to verify that it doesn't cause problems for reflection emitted collectible assemblies before creating a PR for the fix.

**Comment by janvorli** [comment]
Fixed by #22494
